### ISSUE TITLE
Help using handler-sns.rb

### ISSUE THREAD

**Issue by ulutomaz** [sensu-plugins/sensu-plugins-aws#250]
Hi,

would it be possible to get an example with SNS, how to configure it, to be able to send SMS when alerts occur? 

Best,
Tomaz

**Comment by majormoses** [comment]
based on discussion is slack I believe this is likely your problem:
- https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/handler-sns.rb#L17 (uses sdk-v1)
- using latest plugin version (uses `sensu-plugin` 2.x: https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/CHANGELOG.md#700---2017-08-07)
- aws-sdk-v1 conflicts with `sensu-plugin` 2.x because it wants an older json

This I **think** is a duplicate of #240 which is being worked on but will leave it open so that when it is fixed you can validate if your issue is solved following the documentation.

**Comment by eheydrick** [comment]
I'm going to expand the docs for handlers when I update them to AWS-SDK v2. Would love to get your feedback on that @ulutomaz.

**Comment by eheydrick** [comment]
The handlers have been updated to AWS-SDK v2 in 10.0.0. I also added more docs for `handler-sns` to the [README](https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/README.md#usage). There's a few more checks to update before we can remove the v1 dep (see #240) but it's close.

**Comment by ulutomaz** [comment]
I've reconfigured this, but still haven't successfully sent a message. I noticed you're mentioning 2 way of configuring AWS authentication. Can I still put credentials in /etc/sensu/conf.d/sns.json?

Can I use this handler for both? Server side checks and standalone too?

In logs I can't see any relevant info... 

Any suggestion?

**Comment by majormoses** [comment]
There are technically 3 ways to support aws authentication now and only use native methods that aws cli/sdk use. There is documentation for the two recommended solutions:
- IAM role (prefered)
- Use AWS config

The last option which we do not have any documentation for is ENV vars which are a viable option but are too often misused so we do not encourage people to go this route unless they really understand what they are doing and the consequences.

We have documentation [here](https://github.com/sensu-plugins/sensu-plugins-aws#authentication) on authentication.

**Comment by eheydrick** [comment]
Yes you can use this handler for either check type. Can you post the `handler` or `handlers` part of your check definition. You should see something in the server log when the handler is invoked...

`{"timestamp":"2017-10-24T06:21:02.802153+0000","level":"info","message":"handler output","handler":{"type":"pipe","command":"handler-sns.rb","name":"sns"},"output":[]}`

**Comment by majormoses** [comment]
Given the lack of response I am gonna close this out. If you still need help please feel free to re-open.
